cmd/version-reporter/main.go

```diff
@@ -0,0 +1,101 @@
+/*
+version-reporter - report event containing the version of all cacophony software at boot.
+Copyright (C) 2020, The Cacophony Project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package main
+
+import (
+	"log"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/TheCacophonyProject/event-reporter/eventclient"
+)
+
+func main() {
+	err := runMain()
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func runMain() error {
+	log.SetFlags(0) // Removes default timestamp flag
+
+	packageMpedData, err := getInstalledPackages()
+	if err != nil {
+		return err
+	}
+
+	event := eventclient.Event{
+		Timestamp: time.Now(),
+		Type:      "versionData",
+		Details:   packageMpedData,
+	}
+
+	for i := 3; i > 0; i-- {
```

Comment:
any reason to repeat 3 times?

Reply:
Yeah on boot the event-reporter dbus service won't be up right away. So the retry is there to try again if it runs too soon